tests/admin_filters/tests.py

```diff
@@ -1099,3 +1104,43 @@ def test_list_filter_queryset_filtered_by_default(self):
         changelist = modeladmin.get_changelist_instance(request)
         changelist.get_results(request)
         self.assertEqual(changelist.full_result_count, 4)
+
+    def test_should_show_date_hierarchy_choices(self):
```

Comment:
The `admin_filters` app is for testing `django.contrib.admin.templatetags.filters`. It looks like this test could go in `tests/admin_views/test_templatetags.py`. The test passes on master without any changes in this PR, correct? Please send it as a separate pull request. (without the changes to `admin_list.py`)  We don't like to mix additional test coverage with code changes since it gives the impression that behavior is changing as a result of those changes.